workflow/metrics/metrics.go

```diff
@@ -86,33 +88,44 @@ func (m Metrics) allMetrics() []prometheus.Metric {
 	return allMetrics
 }
 
-func (m Metrics) WorkflowAdded(phase v1alpha1.NodePhase) {
+func (m *Metrics) WorkflowAdded(key string, phase v1alpha1.NodePhase) {
+	if m.workflows[key] {
+		return
+	}
 	if _, ok := m.workflowsByPhase[phase]; ok {
+		m.workflows[key] = true
```

Comment:
move one line higher?

Reply:
Yes sorry, this was lingering from when I used to have the check on the `if` statement